lib/source-code/token-store/utils.js

```diff
@@ -77,13 +77,13 @@ exports.getLastIndex = function getLastIndex(tokens, indexMap, endLoc) {
     }
     if ((endLoc - 1) in indexMap) {
         const index = indexMap[endLoc - 1];
-        const token = (index >= 0 && index < tokens.length) ? tokens[index] : null;
+        const token = tokens[index];
 
         /*
          * For the map of "comment's location -> token's index", it points the next token of a comment.
          * In that case, -1 is necessary.
          */
-        if (token && token.range[1] > endLoc) {
+        if (!token || token.range[1] > endLoc) {
```

Comment:
If `token` is undefined, then `tokens` is an empty list and `index` is `0`. In this case, the returned cursor index should point before the first token, at `-1`.
Refs:

- [BackwardTokenCursor](https://github.com/eslint/eslint/blob/27806358b5e1c4d37b63b1c61595e86ff03b5b42/lib/source-code/token-store/backward-token-cursor.js#L34)
- [BackwardTokenCommentCursor](https://github.com/eslint/eslint/blob/27806358b5e1c4d37b63b1c61595e86ff03b5b42/lib/source-code/token-store/backward-token-comment-cursor.js#L35)
- [ForwardTokenCursor](https://github.com/eslint/eslint/blob/27806358b5e1c4d37b63b1c61595e86ff03b5b42/lib/source-code/token-store/forward-token-cursor.js#L35)

Reply:
Same comment as above. :)